### PR TITLE
Use Github release automation

### PR DESCRIPTION
**Pull request by mattwynne**
### 🤔 What's changed?

* Added workflows to release to sonatype and create a GitHub release on push to a `release/*` branch
* Updated release docs
* Modified `Makefile` to no longer use docker or keybase secrets, and push to release branch instead.

### ⚡️ What's your motivation? 

Run releases transparently using GitHub actions.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, docuemntation etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

I'm curious whether we need to do anything about the `-P-examples -P-compatibility` switches which were previously being sent to the `mvn deploy` command. I've not reflected those in the `maven-profiles` argument, but should I? What does the `-` prefix mean?

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

### CODE REVIEW

**Comment by mattwynne** [comment]
OK the Release environment is all set up - I think we should be able to run this whenever you're ready @mpkorstanje

File: .github/workflows/release-github.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+on:
+  push:
+    branches: [release/*]
+
+jobs:
+  pre-release-check:
+    uses: cucumber/.github/.github/workflows/prerelease-checks.yml@main
+
+  create-github-release:
+    name: Create GitHub Release and Git tag
```
Comment (mpkorstanje) [inline .github/workflows/release-github.yml:12]: This uses the changelog tool. Does the current changelog format work with that?

Comment (mattwynne) [reply .github/workflows/release-github.yml:12]: Good question. I'll test it.

Comment (mattwynne) [reply .github/workflows/release-github.yml:12]: Yep, it [looks like this CHANGELOG trips up the tools' parser](https://github.com/cucumber/cucumber-jvm/runs/5588154935?check_suite_focus=true) at the moment.

Although this feature (creating a GitHub release) is nice, it's not essential. Why don't we remove it for now, and we can work on upgrading the changelog tool / tidying this repo's changelog to make them compatible as a separate piece of work?

Comment (mpkorstanje) [reply .github/workflows/release-github.yml:12]: Sounds good to me.

**Comment by mpkorstanje** [comment]
Doesn't work perfectly yet:

https://github.com/cucumber/cucumber-jvm/actions/runs/2010145889

**Comment by mpkorstanje** [comment]
Branch protection rules were missing.

File: .github/workflows/release-java.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  pre-release-check:
+    uses: cucumber/.github/.github/workflows/prerelease-checks.yml@main
+
+  test-java:
+    uses: ./.github/workflows/build.yml
+
+  publish-mvn:
+    name: Publish Maven Package
+    needs: [pre-release-check, test-java]
```
Comment (mpkorstanje) [inline .github/workflows/release-java.yml:16]: This should have referenced `test`, not `test-java`.

**Comment by mpkorstanje** [comment]
This looks promising:

https://github.com/cucumber/cucumber-jvm/actions/runs/2010286622

**Comment by mpkorstanje** [comment]
Nope.

```
2022-03-20T00:58:58.2939863Z [36;1m***[0m
2022-03-20T00:58:58.2941620Z [36;1m	***" | gpg --batch --import --no-tty[0m
2022-03-20T00:58:58.2942282Z [36;1mecho "hello world" > temp.txt[0m
2022-03-20T00:58:58.2942920Z [36;1mgpg --detach-sig \[0m
2022-03-20T00:58:58.2943508Z [36;1m  --yes -v \[0m
2022-03-20T00:58:58.2944114Z [36;1m  --output=/dev/null \[0m
2022-03-20T00:58:58.2944759Z [36;1m  --pinentry-mode loopback \[0m
2022-03-20T00:58:58.2945465Z [36;1m  --passphrase "***" temp.txt[0m
2022-03-20T00:58:58.2946093Z [36;1mrm temp.txt[0m
2022-03-20T00:58:58.3004380Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2022-03-20T00:58:58.3005068Z env:
2022-03-20T00:58:58.3005753Z   JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.14-1/x64
2022-03-20T00:58:58.3006458Z ##[endgroup]
2022-03-20T00:58:58.3182025Z /home/runner/work/_temp/48a9bdca-8775-4c3a-ba18-2702674db02b.sh: line 4: syntax error near unexpected token `)'
2022-03-20T00:58:58.3203797Z ##[error]Process completed with exit code 2.
```

**Comment by mpkorstanje** [comment]
Superseded by:

https://github.com/cucumber/cucumber-jvm/pull/2502

**Pull request merged**